tests/maths/correlation/test_pearson.py

```diff
@@ -51,6 +52,17 @@ def test_pearson_nan_result():
     assert pearson(x, y) == approx(0.6324555320)
 
 
+@mark.parametrize('A', [load_iris().data, load_diabetes().data], ids=['iris', 'diabetes'])
```

Comment:
Nice!

Reply:
If we're going to `parametrize` this, should we add in some of the other sklearn datasets as well?